lib/kubernetes-deploy/kubernetes_resource/job.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  class Job < KubernetesResource
+    TIMEOUT = 10.minutes
+
+    def deploy_succeeded?
+      # Don't block deploys for long running jobs,
+      # Instead report success when there is at least 1 active
+      return false unless deploy_started?
+      done? || running?
+    end
+
+    def deploy_failed?
+      return false unless deploy_started?
+      return false unless @instance_data.dig("spec", "backoffLimit").present?
+      (@instance_data.dig("status", "failed") || 0) >= @instance_data.dig("spec", "backoffLimit")
+    end
+
+    def status
+      if !exists?
+        super
```

Comment:
This is a rather minor point, but it applies to all these resource PRs:  I noticed in 🎩 that `super` results in "Unknown", whereas some of our resources print "Not found" for the same thing, which is inconsistent. Do you have an opinion on which is better? Perhaps the solution is to change the superclass to "Not found" too.

Reply:
From a quick grep it looks like only `Service` uses 'Not found'. I don't have strong feeling on which is better, but I suspect fixing service would be easier.

Reply:
I have a slight preference for "Not found" because it more directly reflects what we actually did to reach this conclusion. Don't need to fix in this PR though.

Reply:
Created an issue https://github.com/Shopify/kubernetes-deploy/issues/321